models/hiking.php

```diff
@@ -22,20 +22,25 @@ public function constructFrowRow($row)
         $this->m_id = $row["id"];
         $this->m_name = $row["name"];
         $this->m_distance = $row["distance"];
+        $this->m_difficulty = $row["difficulty"];
         $this->m_duration = $row["duration"];
         $this->m_height_difference = $row["height_difference"];
     // Initialize object from a database row
   }
 
     public function save()
     {
-        echo "Implement function save()";
+
+       $request = "UPDATE" . $this->m_table_name . "SET name=" . $this->m_name . ", difficulty=" . $this->m_difficulty . ", distance=" . $this->m_distance . ", duration=" . $this->m_duration . ", height_difference=" . $this->m_height_difference. " WHERE id=" . $this->m_id;
+       $pdo = new PDO('mysql:host=localhost;dbname=' . $this->m_database_name . ';charset=utf8', "root", "root");
+       $pdo->query($request);
+       echo $this->m_name;
     }
 
     public function find($id)
     {
-        $request = "SELECT * from " . $this->m_table_name . " WHERE id=" . $id;
-        $pdo = new PDO('mysql:host=localhost;dbname=' . $this->m_database_name, "root", "root");
+        $request = "SELECT * FROM " . $this->m_table_name . " WHERE id=" . $id;
+        $pdo = new PDO('mysql:host=localhost;dbname=' . $this->m_database_name . ';charset=utf8', "root", "root");
```

Comment:
Tu initialise la variable `$pdo` 3 fois exactement de la même façon et donc le même code est répété 3 fois, oublie pas `Less code is better code` ;)